app/services/auth/firebase_auth.py

```
@@ -0,0 +1,386 @@
+    )
+    session.add(new_profile)
+    try:
+        session.commit()
+    except (IntegrityError, DataError, OperationalError, SQLAlchemyError) as e:
+        # Rollback database changes.
+        session.rollback()
+        # cleanup orphaned Firebase user
+        try:
+            auth.delete_user(auth_user.uid)
+            logger.error(f"Database commit failed, cleaned up Firebase user: {e}")
+        except Exception as cleanup_error:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)